fhireval/data/__init__.py

```diff
@@ -29,13 +29,135 @@
+import sys
 
-# DO NOT import this. Just use the load_test_data function. It won't reload if the
-# data has already been loaded.
+from pprint import pformat
+from collections import OrderedDict, defaultdict
+
+from fhireval.test_suite import get_identifier
+
+import pytest
+
+import pdb
+
+# DO NOT import any of these. Only use related functions to ensure that client
+# calls are accessing the proper version in case it's replaced during an intialization
+# call
+
+# products from loading JSON objects
+# Please note that once this module's prep_server is run, this will be modified
+# with valid reference information and will not reflect the original source objects
+# completely
 _resource_data = None
 
+# Identifier=>FHIR ID  (identifier is first identifier in the list)
+_inserted_id_list = {}
+
+# Helpful for ensuring that items with dependencies are correctly loaded/deleted
+# in the right order
+_resource_order = [
+    'Organization',
+    'ResearchStudy',
+    'Patient',
+    'Specimen',
+    'ResearchSubject',
+    'Practitioner',
+    'PractitionerRole',
+    'Condition',
+    'Group',
+    'Observation',
+    'DocumentReference',
+    'Task',
+    'DiagnosticReport',
+    'ConceptMap'
+]
+
+_model_resource_order = [
+    'CodeSystem',
+    'ValueSet',
+    'StructureDefinition'
+]
+
+# Some resources don't actually have an id. Here, we'll provide 
+# the alternate prarameter to search on
+_alternate_search_params = {
+    "SearchParameter": "url"
+}
+
+# Permit the application to bypass reprovision based on flag from user
+_skip_reprovision = False
+
+def skip_reprovision(do_skip=None):
+    """Safely set the skip flag"""
+    global _skip_reprovision
+    if do_skip is not None:
+        _skip_reprovision = do_skip
+
+    return _skip_reprovision
+
+
+def delete_from_response(host, resource_type, response, sleep_til_gone=True):
+    """Delete a resource object based on response from a get
+    
+    Delete a resource based on contents from a get. This will use the object's ID if possible, 
+    which should prevent deleting content with overlapping names or URIs
+
+    :param host:  Fixture referencing the fhirclient object
+    :type host: ncpi-fhir-client
+    :param resource_tye: Which resource type is to be deleted (Patient, ResarchStudy, etc)
+    :type resource_type: String
+    :param response: The record obtained from the fhir server itself with details about what is to be deleted
+    :type response: dict
+    :param sleep_til_gone: When true, the function blocks until the object is truly deleted
+    :type sleep_til_gone: bool
+    """
+    if "resource" in response:
+        id = response['resource']['id']
+
+        # Some systems expect integer IDs, while others expect GUIDs, which obviously aren't convertable to ints
+        try:
+            id = int(id)
+        except:
+            pass
+        delete_result = host.delete_by_record_id(resource_type, id)
+        if delete_result['status_code'] != 409:

```

Comment:
We may not compare it with 409 specifically, but like:

```python
if delete_result['status_code'] >= 300:
```

Reply:
done